### PR TITLE
added rate limit to auth client

### PR DESCRIPTION
**Pull request by chitalian**

### CODE REVIEW

File: valhalla/jawn/src/lib/clients/redisClient.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// import Redis from "ioredis";
+import { Redis } from "@upstash/redis"; // see below for cloudflare and fastly adapters
+
+const redisHost = process.env.REDIS_HOST;
+
+export let redisClient: Redis | null = null;
+if (redisHost) {
+  redisClient = new Redis({
+    url: `${redisHost}:6379`,
+    token: "",
```
Comment (colegottdank) [inline valhalla/jawn/src/lib/clients/redisClient.ts:10]: So if I was following correctly, no token needed here? Secure through VPC?

File: valhalla/jawn/src/middleware/auth.ts [hunk 2]
```diff
@@ -33,6 +49,30 @@ export const authMiddleware = async (
       return;
     }
 
+    const rateLimit = await checkRateLimit(authParams.data?.organizationId);
+    if (rateLimit && !rateLimit.success) {
+      console.error("Rate limit exceeded", authParams.data?.organizationId);
```
Comment (colegottdank) [inline valhalla/jawn/src/middleware/auth.ts:54]: Can we leave console.error to more exceptional errors or else we will just get bombarded with these

Comment (chitalian) [reply valhalla/jawn/src/middleware/auth.ts:54]: yeah

File: valhalla/jawn/src/lib/clients/redisClient.ts (hunk 1 above)
Comment (chitalian) [reply valhalla/jawn/src/lib/clients/redisClient.ts:10]: yep yep

File: valhalla/jawn/src/middleware/auth.ts (hunk 2 above)
Comment (chitalian) [inline valhalla/jawn/src/middleware/auth.ts:54]:
```suggestion
      console.log("Rate limit exceeded", authParams.data?.organizationId);
```

**Pull request merged**